.github/workflows/check-semver.yml

```diff
@@ -12,21 +12,33 @@ concurrency:
 env:
   TOOLCHAIN: nightly-2024-06-01
 
-
 jobs:
   check-semver:
     runs-on: ubuntu-latest
     container:
       image: docker.io/paritytech/ci-unified:bullseye-1.77.0-2024-04-10-v20240408
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          ref: ${{ github.head_ref }} # Check out the actual changes - not the merge commit.
+          fetch-depth: 0
 
       - name: extra git setup
         env:
-          BASE: ${{ github.event.pull_request.base.sha }}
+          BASE_BRANCH: ${{ github.base_ref }}
         run: |
+          set -eux
           git config --global --add safe.directory '*'
+          echo "hi"
+          git fetch --no-tags --no-recurse-submodules --depth=1 origin ${{ github.event.pull_request.base.sha }}
+          git fetch --no-tags --no-recurse-submodules --depth=1 origin master
+          echo "hi2 'git merge-base origin/master HEAD --fork-point'"
+          git merge-base --fork-point origin/$BASE_BRANCH
```

Comment:
maybe the checked out branch isn't as expected - have you tried passing the base.sha here after fetching them both instead of using `--fork-point`?
This should be equivalent, but it doesn't care what is checked out:
```suggestion
          git merge-base ${{ github.event.pull_request.base.sha }} origin/$BASE_BRANCH
```

Reply:
Somehow the `merge-base` command just always reports an error without even telling me the error message. Very confusing...

![Screenshot 2024-08-15 at 17 00 16](https://github.com/user-attachments/assets/e4f8ed1f-ce10-4834-85a7-a794e2a6d1f6)

Reply:
ah wait, the fetch depth is only one, that means that if your branch is two or more commits behind it can't find a common ancestor and it will fail with no error message

Reply:
I think it worked now.  

Got it from this code search: https://github.com/search?q=%22git+show-branch+--merge-base%22+path%3A*.yml&type=code  
GH code search never let me down 😆